lang/cs/Org.Apache.REEF.Network/Group/Topology/FlatTopology.cs

```diff
@@ -115,9 +123,12 @@ public IConfiguration GetTaskConfiguration(string taskId)
             }
             else if (OperatorSpec is ReduceOperatorSpec<T>)
             {
-                ReduceOperatorSpec<T> reduceSpec = OperatorSpec as ReduceOperatorSpec<T>;
-                confBuilder.BindImplementation(typeof(IReduceFunction<T>), reduceSpec.ReduceFunction.GetType());
-                
+                var reduceSpec = OperatorSpec as ReduceOperatorSpec<T>;
+                confBuilder.AddConfiguration(reduceSpec.PipelineDataConverter.GetConfiguration());
+                confBuilder.BindImplementation(typeof(IPipelineDataConverter<T>), reduceSpec.PipelineDataConverter.GetType())
```

Comment:
We need to change spec to put type only. After I work it out, we can change. It is wired to do this way.